### PR TITLE
[ao] quantize.py fixing public v private

### PR DESCRIPTION
**Pull request by HDCharles**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #90570
* #87885
* #87883
* __->__ #87521

Summary: made _register_activation_post_process_hook, _add_observer,
_get_unique_devices_, _get_observer_dict private

Test Plan: python test/test_public_bindings.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D40709277](https://our.internmc.facebook.com/intern/diff/D40709277)

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel

### CODE REVIEW

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40709277).

File: test/allowlist_for_publicAPI.json [hunk 1]
```diff
@@ -780,7 +780,7 @@
     "get_default_static_quant_module_mappings",
     "get_dynamic_quant_module_class",
     "get_fuser_method",
-    "get_observer_dict",
+    "_get_observer_dict",
```
Comment (anjali411) [inline test/allowlist_for_publicAPI.json:783]: you should be able to remove _get_observer_dict from the list, I think?

File: test/allowlist_for_publicAPI.json [hunk 2]
```diff
@@ -891,7 +891,7 @@
     "add_observer_",
     "add_quant_dequant",
     "convert",
-    "get_observer_dict",
+    "_get_observer_dict",
```
Comment (anjali411) [inline test/allowlist_for_publicAPI.json:894]: same as above

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40709277).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40709277).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40709277).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40709277).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40709277).

**Review by jcaip — DISMISSED** [review]

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40709277).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40709277).

**Review by jerryzh168 — APPROVED** [review]

**Pull request closed without merging**